sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs

```diff
@@ -0,0 +1,436 @@
+                    }
+                }
+
+                var logString = $@" *exit code: {pProcess.ExitCode}
+  * cpu_time: {totalProcessorTime} ms
+  * peak_paged_mem: {peakPagedMemory >> 20} MB
+  * peak_working_set: {peakWorkingSet >> 20} MB";
+
+                Log(logString);
+                LogIfFailure(logString);
+
+                if (pProcess.ExitCode != 0)
+                {
+                    Log("Running CBDE: Failure");
+                    LogFailedCbdeRun(pProcess);
+                    Log("Running CBDE: Error dumped");
+                }
+                else
+                {
+                    Log("Running CBDE: Success");
+                    RaiseIssuesFromJSon(c);
+                    Cleanup();
+                    Log("Running CBDE: Issues reported");
+                }
+            }
+        }
+        private void Cleanup()
+        {
+            logStringBuilder.Clear();
+        }
+        private void RaiseIssueFromJToken(JToken token, CompilationAnalysisContext context)
+        {
+            var key = token["key"].ToString();
+            var message = token["message"].ToString();
+            var location = token["location"];
+            var line = Convert.ToInt32(location["l"]);
+            var col = Convert.ToInt32(location["c"]);
+            var file = location["f"].ToString();
+
+            var begin = new LinePosition(line, col);
+            var end = new LinePosition(line, col + 1);
+            var loc = Location.Create(file, TextSpan.FromBounds(0, 0), new LinePositionSpan(begin, end));
+
+            raiseIssue(key, message, loc, context);
+        }
+
+        private void LogFailedCbdeRun(Process pProcess)
+        {
+            StringBuilder failureString = new StringBuilder("CBDE Failure Report :\n  C# souces files involved are:\n");
+            foreach (var fileName in csSourceFileNames)
+            {
+                failureString.Append("  - " + fileName + "\n");
+            }
+            // we dispose the StreamWriter to unlock the log file
+            LogIfFailure($"- parsing json file {cbdeJsonOutputPath}");
+            failureString.Append("  content of stderr is:\n" + pProcess.StandardError.ReadToEnd());
+            failureString.Append("  content of the CBDE handler log file is :\n" + logStringBuilder.ToString());
+            Log(failureString.ToString());
+            Console.Error.WriteLine($"Error when executing CBDE, more details in {cbdeProcessSpecificPath}");
+        }
+
+        private void RaiseIssuesFromJSon(CompilationAnalysisContext context)
+        {
+            string jsonFileContent;
+            List<List<JObject>> jsonIssues;
+            LogIfFailure($"- parsing json file {cbdeJsonOutputPath}");
+            try
+            {
+                jsonFileContent = File.ReadAllText(cbdeJsonOutputPath);
+                jsonIssues = JsonConvert.DeserializeObject<List<List<JObject>>>(jsonFileContent);
+            }
+            catch
```

Comment:
what do you want to catch? please add a specific exception and details about the exception cause in the error message